### PR TITLE
feat: show related lessons by primary tag if not in course

### PR DESCRIPTION
**Pull request by zacjones93**
The lesson list on the lesson page will now populate with related lessons via `primary_tag` when no course is present.

This PR depends on the GraphQL API being merged [via this pr](https://github.com/skillrecordings/egghead-rails/pull/5228) 

## no course
no course
## course
course

g chemex

### CODE REVIEW

File: src/components/pages/lessons/collection-lessons-list.tsx [hunk 1]
```diff
@@ -22,16 +22,17 @@ type NextUpListProps = {
 
 const CollectionLessonsList: FunctionComponent<
   React.PropsWithChildren<NextUpListProps>
-> = ({course, currentLessonSlug, progress, onActiveTab}) => {
-  const {lessons} = course
+> = ({course, currentLessonSlug, progress, onActiveTab, lessons}) => {
   const [activeElement, setActiveElement] = React.useState(currentLessonSlug)
   const scrollableNodeRef: any = React.createRef()
 
+  const lessonList = course ? course.lessons : lessons
```
Comment (zacjones93) [inline src/components/pages/lessons/collection-lessons-list.tsx:29]: when course isn't present, display lessons being passed in

File: src/components/player/player-sidebar.tsx [hunk 2]
```diff
@@ -100,28 +101,45 @@ const LessonListTab: React.FC<
...
   }>
 > = ({videoResource, lessonView, onActiveTab}) => {
-  const hidden: boolean = isEmpty(videoResource.collection)
+  const collectionIsEmpty: boolean = isEmpty(videoResource.collection)
+  const {data: lessonsFromTag} = collectionIsEmpty
+    ? trpc.lesson.getAssociatedLessonsByTag.useQuery({
+        tag: videoResource.primary_tag.name,
+        currentLessonSlug: videoResource.slug,
+      })
+    : {data: null}
```
Comment (zacjones93) [inline src/components/player/player-sidebar.tsx:110]: Opt out of making the http request when a course is present

File: src/components/player/player-sidebar.tsx [hunk 3]
```diff
@@ -100,28 +101,45 @@ const LessonListTab: React.FC<
...
+              currentLessonSlug={videoResource.slug}
+            />
+          </div>
+        )}
+        {collectionIsEmpty && videoResource?.primary_tag && (
+          <div className="flex-shrink-0 p-4 border-gray-100 sm:border-b dark:border-gray-800">
+            <TagHeader
+              tag={videoResource.primary_tag}
+              currentLessonSlug={videoResource.slug}
+            />
```
Comment (zacjones93) [inline src/components/player/player-sidebar.tsx:128]: new Tag header that links to tag page

File: src/lib/lessons.ts [hunk 4]
```diff
@@ -180,6 +180,36 @@ export async function loadLesson(
...
+      id
+      slug
+      completed
+      title
+      description
+      duration
+      free_forever
+      path
+    }
+  }
```
Comment (zacjones93) [inline src/lib/lessons.ts:210]: a subset of lesson attributes, didn't want to pull every attribute when this is just for the lesson list

grabbing 20 lessons for the list to populate

File: src/trpc/routers/lesson.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    )
+    .query(async ({input, ctx}) => {
+      const {tag, currentLessonSlug} = input
+
+      let data = await loadAssociatedLessonsByTag(tag)
+      const filteredData = currentLessonSlug
+        ? data.filter(
+            (lesson: {slug: string}) => lesson.slug !== currentLessonSlug,
+          )
+        : data
```
Comment (zacjones93) [inline src/trpc/routers/lesson.ts:21]: filtering out the current lesson so only other related lessons are shown

File: src/components/pages/lessons/collection-lessons-list.tsx [hunk 6]
```diff
@@ -120,7 +121,7 @@ const CollectionLessonsList: FunctionComponent<
     }
   }, [activeElement, setActiveElement, currentLessonSlug])
 
-  return lessons ? (
+  return lessonList ? (
```
Comment (joelhooks) [inline src/components/pages/lessons/collection-lessons-list.tsx:124]: i've always used "list" postfix for ui components and plural for arrays of data

**Review by joelhooks — APPROVED** [review]

**Pull request merged**